app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentVisitsHeaderViewHolder.kt

```diff
@@ -17,18 +37,62 @@ import org.mozilla.fenix.utils.view.ViewHolder
  * interactions.
  */
 class RecentVisitsHeaderViewHolder(
-    view: View,
+    val view: View,
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
     private val interactor: RecentVisitsInteractor
-) : ViewHolder(view) {
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
-    init {
-        val binding = RecentVisitsHeaderBinding.bind(view)
-        binding.showAllButton.setOnClickListener {
-            interactor.onHistoryShowAllClicked()
+    @Composable
+    override fun Content() {
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+        Column {
+            Spacer(modifier = Modifier.height(40.dp))
+
+            RecentVisitsHeader { interactor.onHistoryShowAllClicked() }
+
+            Spacer(Modifier.height(16.dp))
         }
     }
 
     companion object {
         const val LAYOUT_ID = R.layout.recent_visits_header
     }
 }
+
+@Composable
+private fun RecentVisitsHeader(
+    showAllClick: () -> Unit
+) {
+    val annotatedText = AnnotatedString(text = stringResource(id = R.string.recent_tabs_show_all))
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        SectionHeader(
+            text = stringResource(R.string.history_metadata_header_2),
+            modifier = Modifier
+                .weight(1f, true)
```

Comment:
```suggestion
                .weight(1f)
```
No need for the `true` since that's the default value anyways.